include/pthread.h

```diff
@@ -0,0 +1,376 @@
+#ifndef _PTHREAD_H
+#define _PTHREAD_H
+
+/* FIXME: This should probably live somewhere else but Zephyr doesn't
```

Comment:
Ooh, good point.  Yeah, that's going to be annoying to fix.  The structs are compatible, but the double-definition will be a warning for sure.